packages/core/src/template-compiler-common.ts

```diff
@@ -224,7 +224,20 @@ export class TemplateCompiler {
 }
 
 function matchesSourceFile(filename: string) {
-  return /(htmlbars-inline-precompile|ember-cli-htmlbars)\/(index|lib\/require-from-worker)(\.js)?$/.test(filename);
+  // this respects ember-cli-htmlbars\lib\require-from-worker.js and
+  // ember-cli-htmlbars/lib/require-from-worker.js
+  let matches = [
```

Comment:
I don't think you need to construct this lookup structure on each invocation of `matchesSourceFile` rather a frozen array could be kept at  module scape, and reused in this function. That would prevent having to reconstruct.

Reply:
A alternative would be to create the matching structure from glob via picomatch https://github.com/micromatch/picomatch as some may consider a glob to be more readable. The downside is an extra dependency..

Reply:
Moved this outside the function scope and into the module's scope.